### ISSUE TITLE
Pytorch scheduler .step function needs to be moved for newer pytorch version

### ISSUE THREAD

**Issue by derEitel** [derEitel/nitorch#14]
The recent pytorch versions move the .step function of the .optim schedulers to the end of the training loop rather than the beginning.
